Complete test layout configuration

This is like a follow up for #17.

Problem to solve: if I move fixtures inside a conftest.py, `import project` does not work.

Local:
```
ImportError while loading conftest '/Users/iacopo/Dropbox/dev/BIOINFO/RCSB_Sync/tests/conftest.py'.
tests/conftest.py:9: in <module>
    from project import Project
E   ModuleNotFoundError: No module named 'project'
```

CI
```
ImportError while loading conftest '/home/runner/work/RCSB-Sync/RCSB-Sync/tests/conftest.py'.
[8](https://github.com/iacopy/RCSB-Sync/actions/runs/3255920731/jobs/5345739953#step:7:9)
tests/conftest.py:9: in <module>
[9](https://github.com/iacopy/RCSB-Sync/actions/runs/3255920731/jobs/5345739953#step:7:10)
    from project import Project
[10](https://github.com/iacopy/RCSB-Sync/actions/runs/3255920731/jobs/5345739953#step:7:11)
E   ModuleNotFoundError: No module named 'project'
```

--- a/pytest.ini
+++ b/pytest.ini
@@ -1,3 +1,4 @@
 [pytest]
+pythonpath = src
 testpaths = .
 addopts = --doctest-modules --benchmark-disable

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -43,6 +43,22 @@ def project_with_files(empty_project):
     return empty_project
 
 
+def result_set(ids):
+    """Return a json string with a list of ids."""
+    return '{"result_set": [' + ', '.join([f'{{"identifier": "{id_}"}}' for id_ in ids]) + ']}'
+
+
+def make_search_response(ids):
+    """Return an OK search response object with a list of ids."""
+    return responses.Response(
+        method='GET',
+        url=f'{SEARCH_ENDPOINT_URI}',
+        body=result_set(ids),
+        status=200,
+        content_type="application/json",
+    )
+
+
 @pytest.fixture
 def mocked_responses():
     """Return a mocked responses object."""
@@ -53,16 +69,26 @@ def mocked_responses():
 @pytest.fixture
 def remote_server(mocked_responses):
     """Return a mocked remote server with ids."""
-    mocked_responses.get(
-        url=f'{SEARCH_ENDPOINT_URI}',
-        body='{"result_set": [{"identifier": "hs01"}, {"identifier": "hs02"}]}',
-        status=200,
-        content_type="application/json",
+    # Add responses to the mocked server for the queries.
+    mocked_responses.add(
+        make_search_response(['hs01', 'hs02'])
     )
-    mocked_responses.get(
-        url=f'{SEARCH_ENDPOINT_URI}',
-        body='{"result_set": [{"identifier": "rn01"}, {"identifier": "rn02"}]}',
-        status=200,
-        content_type="application/json",
+    # Second query.
+    mocked_responses.add(
+        make_search_response(['rn01', 'rn02'])
+    )
+    return mocked_responses
+
+
+@pytest.fixture
+def remote_server_changed(mocked_responses):
+    """Return a mocked remote server with an id removed."""
+    # Add responses to the mocked server for the queries.
+    mocked_responses.add(
+        make_search_response(['hs01', 'hs02', 'hs03'])
+    )
+    # Second query.
+    mocked_responses.add(
+        make_search_response(['rn01'])
     )
     return mocked_responses

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,0 +1,68 @@
+"""
+Fixtures.
+"""
+from pathlib import Path
+
+import pytest
+import responses
+
+from project import Project
+from rcsbids import SEARCH_ENDPOINT_URI
+
+# Fake json queries
+HS_QUERY = '{"query": {"query_type": "terminal", "search_type": "text", "value": "Homo sapiens"}}'
+RN_QUERY = '{"query": {"query_type": "terminal", "search_type": "text", "value": "Rattus norvegicus"}}'
+
+
+@pytest.fixture
+def empty_project(tmp_path):
+    """Return an empty project."""
+    # Create a project in a temporary directory.
+    queries = tmp_path / 'queries'
+    queries.mkdir()
+    # Create a fake query file.
+    query_file1 = queries / 'Homo_sapiens.json'
+    query_file2 = queries / 'Rattus_norvegicus.json'
+    query_file1.write_text(HS_QUERY)
+    query_file2.write_text(RN_QUERY)
+    return Project(tmp_path)
+
+
+@pytest.fixture
+def project_with_files(empty_project):
+    """Return a project with some downloaded files."""
+    # add pdb.gz files inside the project directory
+    hs01 = Path(empty_project.data_dir, 'hs01.pdb.gz')
+    hs02 = Path(empty_project.data_dir, 'hs02.pdb.gz')
+    rn01 = Path(empty_project.data_dir, 'rn01.pdb.gz')
+    rn02 = Path(empty_project.data_dir, 'rn02.pdb.gz')
+    hs01.write_text('hs01', encoding='ascii')
+    hs02.write_text('hs02', encoding='ascii')
+    rn01.write_text('rn01', encoding='ascii')
+    rn02.write_text('rn02', encoding='ascii')
+    return empty_project
+
+
+@pytest.fixture
+def mocked_responses():
+    """Return a mocked responses object."""
+    with responses.RequestsMock() as rsps:
+        yield rsps
+
+
+@pytest.fixture
+def remote_server(mocked_responses):
+    """Return a mocked remote server with ids."""
+    mocked_responses.get(
+        url=f'{SEARCH_ENDPOINT_URI}',
+        body='{"result_set": [{"identifier": "hs01"}, {"identifier": "hs02"}]}',
+        status=200,
+        content_type="application/json",
+    )
+    mocked_responses.get(
+        url=f'{SEARCH_ENDPOINT_URI}',
+        body='{"result_set": [{"identifier": "rn01"}, {"identifier": "rn02"}]}',
+        status=200,
+        content_type="application/json",
+    )
+    return mocked_responses

--- a/tests/test_project.py
+++ b/tests/test_project.py
@@ -3,73 +3,13 @@ Unit tests for the project module.
 """
 # Standard Library
 import os
-from pathlib import Path
 
 # 3rd party
 import pytest
-import responses
 
 # My stuff
 from project import Project
 from rcsbids import SEARCH_ENDPOINT_URI
-
-# Fake json queries
-HS_QUERY = '{"query": {"query_type": "terminal", "search_type": "text", "value": "Homo sapiens"}}'
-RN_QUERY = '{"query": {"query_type": "terminal", "search_type": "text", "value": "Rattus norvegicus"}}'
-
-
-@pytest.fixture
-def empty_project(tmp_path):
-    """Return an empty project."""
-    # Create a project in a temporary directory.
-    queries = tmp_path / 'queries'
-    queries.mkdir()
-    # Create a fake query file.
-    query_file1 = queries / 'Homo_sapiens.json'
-    query_file2 = queries / 'Rattus_norvegicus.json'
-    query_file1.write_text(HS_QUERY)
-    query_file2.write_text(RN_QUERY)
-    return Project(tmp_path)
-
-
-@pytest.fixture
-def project_with_files(empty_project):
-    """Return a project with some downloaded files."""
-    # add pdb.gz files inside the project directory
-    hs01 = Path(empty_project.data_dir, 'hs01.pdb.gz')
-    hs02 = Path(empty_project.data_dir, 'hs02.pdb.gz')
-    rn01 = Path(empty_project.data_dir, 'rn01.pdb.gz')
-    rn02 = Path(empty_project.data_dir, 'rn02.pdb.gz')
-    hs01.write_text('hs01', encoding='ascii')
-    hs02.write_text('hs02', encoding='ascii')
-    rn01.write_text('rn01', encoding='ascii')
-    rn02.write_text('rn02', encoding='ascii')
-    return empty_project
-
-
-@pytest.fixture
-def mocked_responses():
-    """Return a mocked responses object."""
-    with responses.RequestsMock() as rsps:
-        yield rsps
-
-
-@pytest.fixture
-def remote_server(mocked_responses):
-    """Return a mocked remote server with ids."""
-    mocked_responses.get(
-        url=f'{SEARCH_ENDPOINT_URI}',
-        body='{"result_set": [{"identifier": "hs01"}, {"identifier": "hs02"}]}',
-        status=200,
-        content_type="application/json",
-    )
-    mocked_responses.get(
-        url=f'{SEARCH_ENDPOINT_URI}',
-        body='{"result_set": [{"identifier": "rn01"}, {"identifier": "rn02"}]}',
-        status=200,
-        content_type="application/json",
-    )
-    return mocked_responses
 
 
 def test_project_non_existing_directory():
@@ -119,6 +59,7 @@ def test_removed_handling(project_with_files, mocked_responses):
     """
     Test that the remote-removed ids are handled properly.
     """
+    # NB: this code should be refactored, since is a duplicate of a part of remote_server fixture.
     mocked_responses.get(
         url=f'{SEARCH_ENDPOINT_URI}',
         body='{"result_set": [{"identifier": "hs01"}, {"identifier": "hs02"}]}',

